### PR TITLE
For #9072: Add new test for Home button

### PR DESCRIPTION
**Pull request by AaronMT**
Adds a test for the toolbar Home button.

Note: Issue: https://github.com/mozilla-mobile/firefox-ios/issues/9083 – not sure if this should hold on this issue

### CODE REVIEW

File: XCUITests/HomeButtonTests.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class HomeButtonTests: BaseTestCase {
+
+    override func tearDown() {
+        XCUIDevice.shared.orientation = .portrait
+        super.tearDown()
+    }
+
+    func testGoHome() throws {
+        navigator.openURL("https://www.google.com", waitForLoading: true)
```
Comment (isabelrios) [inline XCUITests/HomeButtonTests.swift:15]: Could we use a local server site? like: https://github.com/mozilla-mobile/firefox-ios/blob/main/XCUITests/TopTabsTest.swift#L12
or https://github.com/mozilla-mobile/firefox-ios/blob/main/XCUITests/TopTabsTest.swift#L26

File: XCUITests/HomeButtonTests.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        waitForExistence(app.buttons["TabToolbar.homeButton"], timeout: 5)
+        XCTAssertTrue(app.buttons["TabToolbar.homeButton"].exists)
+        app.buttons["TabToolbar.homeButton"].tap()
+        navigator.nowAt(NewTabScreen)
+        navigator.performAction(Action.CloseURLBarOpen)
+        waitForTabsButton()
+
+        XCTAssertEqual(app.buttons["TabToolbar.homeButton"].label, "Search")
+        navigator.openURL("https://www.twitter.com", waitForLoading: true)
```
Comment (isabelrios) [inline XCUITests/HomeButtonTests.swift:24]: Same as above about using a testing site. Sometime this site takes to long to load on CI and cause flakiness

File: XCUITests/HomeButtonTests.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+class HomeButtonTests: BaseTestCase {
```
Comment (isabelrios) [inline XCUITests/HomeButtonTests.swift:7]: When a new test is create it is enabled by default in all schemes. Could you please disable it from the SmokeTest or the FullFuncional test plan according to where it is run manually? 

Thanks!

**Review by isabelrios — CHANGES_REQUESTED** [review]
Thanks @AaronMT ! It looks good :) there are a few things that I would change though as you will see in the comments.

File: XCUITests/HomeButtonTests.swift (hunk 1 above)
Comment (AaronMT) [reply XCUITests/HomeButtonTests.swift]: Done.

File: XCUITests/HomeButtonTests.swift (hunk 2 above)
Comment (AaronMT) [reply XCUITests/HomeButtonTests.swift]: Done.

File: XCUITests/HomeButtonTests.swift (hunk 3 above)
Comment (AaronMT) [reply XCUITests/HomeButtonTests.swift:7]: Oh this is good to know for new tests. I disabled everywhere except `Fennec_Enterprise_XCUITests`. Let me know if the is correct.

Comment (isabelrios) [reply XCUITests/HomeButtonTests.swift:7]: Yes, that should be fine, thanks!

**Review by isabelrios — APPROVED** [review]
Thanks for the changes!

**Pull request merged**